MAINTAINERS.md

```diff
@@ -1,61 +0,0 @@
-# For maintainers only
-Minio Java SDK uses [gradle](https://gradle.org/) build system.
-
-## Responsibilities
-Go through [Maintainer Responsibility Guide](https://gist.github.com/abperiasamy/f4d9b31d3186bbd26522).
```

Comment:
MAINTAINERS.md is still needed but we need to reduce the scope of this doc by removing the release related documentation.